### PR TITLE
Raven db 5595

### PR DESCRIPTION
**Pull request by Danielle9897**

### CODE REVIEW

File: src/Raven.Studio/typescript/common/helpers/graph/rangeAggregator.ts [hunk 1]
```diff
@@ -3,73 +3,94 @@
...
-        // after all
-        if (start > this.aggregation.last().end) {
-            if (this.aggregation.last().end !== start) {
-                this.aggregation.push({
-                    start: this.aggregation.last().end,
-                    end: start,
-                    value: 0
-                });
-            }
+        this.items = inputItems.map((x) => { return { startTime: x[0].getTime(), endTime: x[1].getTime() } });
```
Comment (ml054) [inline src/Raven.Studio/typescript/common/helpers/graph/rangeAggregator.ts]: remove { return

Comment (ml054) [reply src/Raven.Studio/typescript/common/helpers/graph/rangeAggregator.ts]: inputItems.map((x) => ({ startTime: x[0].getTime(), endTime: x[1].getTime() }));

you have to enclose object in () to make it work.

File: src/Raven.Studio/typescript/viewmodels/database/indexes/indexPerformance.ts [hunk 2]
```diff
@@ -105,7 +106,8 @@ class metrics extends viewModelBase {
     static readonly colors = {
         axis: "#546175",
         gaps: "#ca1c59",
-        brushChartColor: "#008cc9",
+        brushChartColor: "#37404b",
+        brushChartFrameColor: "#008cc9",
```
Comment (ml054) [inline src/Raven.Studio/typescript/viewmodels/database/indexes/indexPerformance.ts]: rename: brushChartStrokeColor

File: src/Raven.Studio/typescript/viewmodels/database/indexes/indexPerformance.ts [hunk 3]
```diff
@@ -330,40 +335,58 @@ class metrics extends viewModelBase {
...
         this.brushSection = document.createElement("canvas");
         this.brushSection.width = this.totalWidth;
         this.brushSection.height = metrics.brushSectionHeight;
 
-        const context = this.brushSection.getContext("2d");
-
+        this.gapFinder = new gapFinder(timeRanges, metrics.minGapSize);
         this.xBrushTimeScale = this.gapFinder.createScale(this.totalWidth, 0);
+        
+        this.yBrushValueScale = d3.scale.linear().domain([0, maxConcurrentIndexes]).range([0, metrics.brushSectionIndexesWorkHeight]);
```
Comment (ml054) [inline src/Raven.Studio/typescript/viewmodels/database/indexes/indexPerformance.ts]: reformat to:
d3.scale.linear()
.domain([0, maxConcurrentIndexes])
.range([0, metrics.brushSectionIndexesWorkHeight]);

File: src/Raven.Studio/typescript/viewmodels/database/indexes/indexPerformance.ts [hunk 4]
```diff
@@ -330,40 +335,58 @@ class metrics extends viewModelBase {
...
 
-        for (var i = 0; i < collapsedTimeRanges.length; i++) {
-            const currentRange = collapsedTimeRanges[i];
-            const x1 = this.xBrushTimeScale(currentRange[0]);
-            const x2 = this.xBrushTimeScale(currentRange[1]);
-            context.fillRect(x1, 18, x2 - x1, 10);
-            context.strokeRect(x1, 18, x2 - x1, 10);
+        // Draw area chart showing indexes work
+        let x1: number, x2: number, y0: number = 0, y1: number;
+        for (var i = 0; i < workData.length - 1; i++) {
```
Comment (ml054) [inline src/Raven.Studio/typescript/viewmodels/database/indexes/indexPerformance.ts]: var -> let

File: src/Raven.Studio/typings/_studio/dto.d.ts [hunk 5]
```diff
@@ -157,4 +157,14 @@ interface aggregatedRange {
     start: number;
     end: number;
     value: number;
+}
+
+interface indexesWorkData {
```
Comment (ml054) [inline src/Raven.Studio/typings/_studio/dto.d.ts:162]: move those interfaces to viewmodel class - they are private, so we don't pollute dto.d.ts.

**Review by ml054 — CHANGES_REQUESTED** [review]

File: src/Raven.Studio/typescript/common/helpers/graph/rangeAggregator.ts (hunk 1 above)
Comment (Danielle9897) [reply src/Raven.Studio/typescript/common/helpers/graph/rangeAggregator.ts]: done

File: src/Raven.Studio/typescript/viewmodels/database/indexes/indexPerformance.ts (hunk 2 above)
Comment (Danielle9897) [reply src/Raven.Studio/typescript/viewmodels/database/indexes/indexPerformance.ts]: done

File: src/Raven.Studio/typescript/viewmodels/database/indexes/indexPerformance.ts (hunk 3 above)
Comment (Danielle9897) [reply src/Raven.Studio/typescript/viewmodels/database/indexes/indexPerformance.ts]: done

File: src/Raven.Studio/typescript/viewmodels/database/indexes/indexPerformance.ts (hunk 4 above)
Comment (Danielle9897) [reply src/Raven.Studio/typescript/viewmodels/database/indexes/indexPerformance.ts]: done

File: src/Raven.Studio/typings/_studio/dto.d.ts (hunk 5 above)
Comment (Danielle9897) [reply src/Raven.Studio/typings/_studio/dto.d.ts:162]: Both rangeAggregator.ts & the viewModel class use them

**Comment by ayende** [comment]
Please note that this PR has conflicts, please fix them.

**Comment by ml054** [comment]
@Danielle9897 I will resolve this conflict tomorrow, after I resolve: https://github.com/ravendb/ravendb/pull/1112

**Comment by ml054** [comment]
Ok, ready for merge.

**Pull request merged**